### PR TITLE
[RFC] Rename macro RGB as NVIM_RGB

### PR DESCRIPTION
**Pull request by equalsraf**
Windows headers(wingdi.h/windows.h) already define a macro named RGB().

### CODE REVIEW

**Comment by fwalch** [comment]
LGTM.

**Comment by equalsraf** [comment]
Waiting for Travis to tag RDY

File: src/nvim/macros.h [hunk 1]
```diff
@@ -154,6 +154,6 @@
 /// zero in those cases (-Wdiv-by-zero in GCC).
 #define ARRAY_SIZE(arr) ((sizeof(arr)/sizeof((arr)[0])) / ((size_t)(!(sizeof(arr) % sizeof((arr)[0])))))
 
-#define RGB(r, g, b) ((r << 16) | (g << 8) | b)
+#define NVIM_RGB(r, g, b) ((r << 16) | (g << 8) | b)
```
Comment (Pyrohh) [inline src/nvim/macros.h]: What about this instead?
```
#undef RGB
#define RGB(r, g, b) ((r << 16) | (g << 8) | b)
```

**Comment by equalsraf** [comment]
> Although, I wonder, why does Vim's Windows build not have this problem?

@fwalch I'm seeing several #defines for RGB in the vim source, one for gui_mac.c, another for gui_photon.c so it is possible it is in fact using the one from windows.h.

@Pyrohh is right, closing ths.

**Comment by equalsraf** [comment]
@jszakmeister what just happened? I just closed this.

**Comment by fwalch** [comment]
@equalsraf it's not really @jszakmeister, but waffle.io.. Seems to happen sometimes :-/

File: src/nvim/macros.h (hunk 1 above)
Comment (aktau) [inline src/nvim/macros.h]: Or what about:

```c
static inline uint32_t rgb(uint8_t r, uint8_t g, uint8_t b) {
  return (r << 16) | (g << 8) | b;
}
```

Assuming that we're always supplying and expecting those types. Didn't look it up (for which my apologies).

Comment (Pyrohh) [inline src/nvim/macros.h]: I forgot to ask earlier, hence the delay.

Anyways, I'm curious: what's the benefit of using a function in this case as opposed to the macro mentioned above? I'm guessing the function provides extra type checking, which would avoid or at least reduce the possibility of left-shifting to the point of undefined behaviour if a signed type were accidentally supplied. As I said, I'm unsure :P

Comment (aktau) [inline src/nvim/macros.h]: In fact I thought about it and the way we are using the macro we simply can't transform it into a function: it's used to generate constant values in the `color_name_table`. The C compiler wouldn't accept a function there afaik. One would need some C++11 magic for that, which is even uglier.

As a general guideline, if something _can_ be transformed from a macro to a `static inline`, that's great. But as we see, that's not always possible.

So, please forget what I said :).

**Pull request closed without merging**